### PR TITLE
[xharness] Don't report tests whose device were not found if there were no such tests.

### PR DESCRIPTION
**Pull request by rolfbjarne**

### CODE REVIEW

File: tests/xharness/Jenkins.cs [hunk 1]
```diff
@@ -1772,7 +1772,11 @@ void GenerateReportImpl (Stream stream, StreamWriter markdown_summary = null)
 					markdown_summary.Write ($"# Test run in progress: ");
 					markdown_summary.Write (string.Join (", ", list));
 				} else if (failedTests.Any ()) {
-					markdown_summary.Write ($"{failedTests.Count ()} tests failed, {deviceNotFound.Count ()} tests' device not found, {passedTests.Count ()} tests passed.");
+					if (deviceNotFound.Any ()) {
+						markdown_summary.Write ($"{failedTests.Count ()} tests failed, {deviceNotFound.Count ()} tests' device not found, {passedTests.Count ()} tests passed.");
+					} else {
+						markdown_summary.Write ($"{failedTests.Count ()} tests failed, {passedTests.Count ()} tests passed.");
+					}
```
Comment (spouliot) [inline tests/xharness/Jenkins.cs]: DRY :)
```csharp
markdown_summary.Write ($"{failedTests.Count ()} tests failed,");
if (deviceNotFound.Any ())
    markdown_summary.Write ($"{deviceNotFound.Count ()} tests' device not found, ");
markdown_summary.Write ($"{passedTests.Count ()} tests passed.");
```

File: tests/xharness/Jenkins.cs [hunk 2]
```diff
@@ -1856,7 +1860,11 @@ void GenerateReportImpl (Stream stream, StreamWriter markdown_summary = null)
 					writer.Write (string.Join (", ", list));
 					writer.Write (")");
 				} else if (failedTests.Any ()) {
-					writer.Write ($"{failedTests.Count ()} tests failed, {deviceNotFound.Count ()} tests' device not found, {passedTests.Count ()} tests passed");
+					if (deviceNotFound.Any ()) {
+						writer.Write ($"{failedTests.Count ()} tests failed, {deviceNotFound.Count ()} tests' device not found, {passedTests.Count ()} tests passed");
+					} else {
+						writer.Write ($"{failedTests.Count ()} tests failed, {passedTests.Count ()} tests passed");
+					}
```
Comment (spouliot) [inline tests/xharness/Jenkins.cs]: same

**Comment by rolfbjarne** [comment]
Test failure is unrelated

* monotouch-test/iOS 64-bits: https://github.com/xamarin/maccore/issues/1595

**Comment by rolfbjarne** [comment]
@monojenkins backport d16-2

**Pull request merged**